packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -660,7 +660,7 @@ export class Relayer {
     const relayableDeposits: RelayableDeposits = {};
     for (const l1Token of this.whitelistedRelayL1Tokens) {
       this.logger.debug({ at: "Relayer", message: "Checking relays for token", l1Token });
-      const l2Deposits = this.l2Client.getAllDeposits();
+      const l2Deposits = this.l2Client.getAllDepositsForL1Token(l1Token);
```

Comment:
this bit of code encapsulates the bug that the bot had: we are iterating over all `this.whitelistedRelayL1Tokens` and for _each_ pulling out _all_ deposits. Rather, we should only be looking at the deposits for the given `l1Token` to prevent duplication.

Reply:
ahhhhhhh we're so dumb! nice catch

Reply:
This also makes sense why duplicate txns were sending only on relays but not on settlements or disputes